python/interpret_community/common/model_wrapper.py

```diff
@@ -231,6 +232,45 @@ def predict(self, dataset):
         return self._eval_function(dataset)
 
 
+class WrappedClassificationWithoutProbaModel(object):
+    """A class for wrapping a classifier without a predict_proba method.
+
+    Note: the classifier may not output numeric values for its predictions.
+    We generate a trival boolean version of predict_proba
+    """
+
+    def __init__(self, model):
+        """Initialize the WrappedClassificationWithoutProbaModel with the model."""
+        self._model = model
+        # Create a map from classes to index
+        self._classes_to_index = {}
+        for index, i in enumerate(self._model.classes_):
+            self._classes_to_index[i] = index
+        self._num_classes = len(self._model.classes_)
+
+    def predict(self, dataset):
+        """Predict the output using the wrapped regression model.
+
+        :param dataset: The dataset to predict on.
+        :type dataset: DatasetWrapper
+        """
+        return self._model.predict(dataset)
+
+    def predict_proba(self, dataset):
+        """Predict the output probability using the wrapped model.
+
+        :param dataset: The dataset to predict_proba on.
+        :type dataset: DatasetWrapper
+        """
+        predictions = self.predict(dataset)
+        # Generate trivial boolean array for predictions
+        probabilities = np.zeros((predictions.shape[0], self._num_classes))
+        for row_idx, pred_class in enumerate(predictions):
+            class_index = self._classes_to_index[pred_class]
+            probabilities[row_idx, class_index] = 1
+        return probabilities
```

Comment:
One alternative method is to calibrate (possibly [with sklearn](https://scikit-learn.org/stable/modules/generated/sklearn.calibration.CalibratedClassifierCV.html)) so that you always get probabilities. Do we want a different base class that uses this method?

Reply:
Agree, had this thought as well, but I think for now I wanted to do the simpler thing.  The user can add calibration on their own for now by wrapping their model and adding predict_proba.

Reply:
It might be good to add an additional helper class though that the user could eventually use.